WalletWasabi/Blockchain/Transactions/TransactionFactory.cs

```diff
@@ -121,12 +121,13 @@ public TransactionFactory(Network network, KeyManager keyManager, ICoinsView coi
 				}
 			}
 
-			HdPubKey changeHdPubKey;
+			HdPubKey? changeHdPubKey;
 
 			if (payments.TryGetCustomRequest(out DestinationRequest? custChange))
 			{
 				var changeScript = custChange.Destination.ScriptPubKey;
-				changeHdPubKey = KeyManager.GetKeyForScriptPubKey(changeScript);
+				KeyManager.TryGetKeyForScriptPubKey(changeScript, out HdPubKey? hdPubKey);
```

Comment:
Something doesn't look 100% okay here. We know it will return true but looks ummm... Anyway, it is better than before.

Reply:
Do not waste time in this, btw. It was just a comment.